### PR TITLE
Fixed crash + added more tests.

### PR DESCRIPTION
**Pull request by aosyatnik**
### Purpose

[MAGN-9488](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-9488) Create a CBN gives a crash

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.

### Reviewers

@ramramps 

### FYIs

@kronz @riteshchandawar @aparajit-pratap

### CODE REVIEW

File: src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs [hunk 1]
```diff
@@ -580,8 +580,14 @@ internal void TogglePreviewControlAllowance()
 
             if (previewEnabled == false && !PreviewControl.StaysOpen)
             {
-                PreviewControl.TransitionToState(PreviewControl.State.Condensed);
-                PreviewControl.TransitionToState(PreviewControl.State.Hidden);
+                if (PreviewControl.IsExpanded)
+                {
+                    PreviewControl.TransitionToState(PreviewControl.State.Condensed);
```
Comment (ramramps) [inline src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs]: condensed and hidden? Hidden should always make the preview condense and hide. Think about it.

**Comment by ramramps** [comment]
@aosyatnik - One change here. Then LGTM.

File: src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs (hunk 1 above)
Comment (aosyatnik) [inline src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs]: But if we try to hide immediately, when preview is expanded, we will get our crash.
Did I get the question right?

Comment (ramramps) [inline src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs]: Ok. 
FYI : It is not right to transition like this. There has to be an intervention either externally or self.
but for now merge this in.

**Pull request merged**